pkg/storage/pv_test.go

```diff
@@ -0,0 +1,211 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const defaultPersistentVolumeName = "persistentvolume-test"
+
+func TestPullPersistentVolume(t *testing.T) {
+	testCases := []struct {
+		persistentVolumeName string
+		addToRuntimeObjects  bool
+		client               bool
+		expectedErrorText    string
+	}{
+		{
+			persistentVolumeName: defaultPersistentVolumeName,
+			addToRuntimeObjects:  true,
+			client:               false,
+			expectedErrorText:    fmt.Sprintf("PersistentVolume object %s does not exist", defaultPersistentVolumeName),
```

Comment:
This is invalid error. Please add apiClinet == nil validation to pull function and fix return error